src/components/TodoList/TodoList.tsx

```diff
@@ -1,210 +1,102 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useCallback } from 'react';
+import classNames from 'classnames';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { Todo } from '../../types/Todo';
+import { actions } from '../../features/currentTodo';
 
 export const TodoList: React.FC = () => {
-  return (
-    <>
-      <p className="notification is-warning">
-        There are no todos matching current filter criteria
-      </p>
-
-      <table className="table is-narrow is-fullwidth">
-        <thead>
-          <tr>
-            <th>#</th>
+  const dispatch = useAppDispatch();
+  const { todos, currentTodo, filter } = useAppSelector(state => state);
+  const { query, status } = filter;
 
-            <th>
-              <span className="icon">
-                <i className="fas fa-check" />
-              </span>
-            </th>
-
-            <th>Title</th>
-            <th> </th>
-          </tr>
-        </thead>
-
-        <tbody>
-          <tr data-cy="todo">
-            <td className="is-vcentered">1</td>
-            <td className="is-vcentered"> </td>
-
-            <td className="is-vcentered is-expanded">
-              <p className="has-text-danger">delectus aut autem</p>
-            </td>
-
-            <td className="has-text-right is-vcentered">
-              <button data-cy="selectButton" className="button" type="button">
-                <span className="icon">
-                  <i className="far fa-eye" />
-                </span>
-              </button>
-            </td>
-          </tr>
+  const onButtonClick = (todo: Todo) => {
+    dispatch(actions.setTodo(todo));
+  };
 
-          <tr data-cy="todo">
-            <td className="is-vcentered">2</td>
-            <td className="is-vcentered"> </td>
+  const selectList = () => {
+    switch (status) {
+      case 'active':
+        return todos.filter(todo => todo.completed === false);
 
-            <td className="is-vcentered is-expanded">
-              <p className="has-text-danger">quis ut nam facilis et officia qui</p>
-            </td>
+      case 'completed':
+        return todos.filter(todo => todo.completed === true);
 
-            <td className="has-text-right is-vcentered">
-              <button data-cy="selectButton" className="button" type="button">
-                <span className="icon">
-                  <i className="far fa-eye" />
-                </span>
-              </button>
-            </td>
-          </tr>
+      default:
+        return todos;
```

Comment:
```suggestion
        return true;
```
you don't actually return `todos` here - filter callback should return a boolean value, so returned by you `todos` is transformed to a boolean value under the hood - it is always true because it's an array. but better to return `true` explicitly (FYI)